File: api/models/Habitat.js
module.exports = (sequelize, DataTypes) => {
  const Habitat = sequelize.define(
    "Habitat",
    {
      id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
      },
      name: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      description: {
        type: DataTypes.STRING,
        allowNull: true,
      },
      climate: {
        type: DataTypes.STRING,
        allowNull: true,
      },
      temperature: {
        type: DataTypes.FLOAT,
        allowNull: true,
      },
      humidity: {
        type: DataTypes.FLOAT,
        allowNull: true,
      },
    },
    {
      tableName: "Habitat",
      timestamps: false,
    },
  );

  return Habitat;
};

File: app/lib/models/Habitat.dart
class Habitat {
  final int id;
  final String name;
  final String description;
  final String climate;
  final int temperature;
  final int humidity;

  Habitat({
    required this.id,
    required this.name,
    required this.description,
    required this.climate,
    required this.temperature,
    required this.humidity,
  });

  factory Habitat.fromJson(Map<String, dynamic> json) {
    return Habitat(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      climate: json['climate'],
      temperature: json['temperature'],
      humidity: json['humidity'],
    );
  }
}
